hir2/src/dialects/builtin/builders/world.rs

```diff
@@ -38,4 +40,25 @@ impl WorldBuilder {
         let builder = PrimComponentBuilder::new(&mut self.builder, name.span());
         builder(ns, name, ver)
     }
+
+    /// Declare a new world-level module `name`
+    pub fn declare_module(&mut self, name: Ident) -> Result<ModuleRef, Report> {
```

Comment:
Ha! For some reason, I thought that `World` can contain only `Component`s. OTOH it makes sense to link stdlib and tx_kernel as modules.

Reply:
Yeah my thinking is that "plain" modules, e.g. the Miden stdlib or whatever, can be declared/translated as world-level modules (since we can then just use their "real" symbol names when calling procedures exported from them). It makes the structure a bit more fluid, but feels like it is a better way to represent modules that aren't sourced from a Wasm component (e.g. intrinsics).